paddle/fluid/framework/op_registry.cc

```diff
@@ -21,12 +21,16 @@ namespace framework {
 
 std::unique_ptr<OperatorBase> OpRegistry::CreateOp(
     const std::string& type, const VariableNameMap& inputs,
-    const VariableNameMap& outputs, AttributeMap attrs, bool attr_check) {
+    const VariableNameMap& outputs, AttributeMap attrs,
+    const std::vector<std::string>& input_names,
```

Comment:
Although I understand it is difficult here, but there is already `input_names` in the `inputs` and only the order information is missing, but in this case the input information is redundant and not a good design. Can we find a way to implement an efficient `ordered_map` here?

for example:
maybe we can define a `paddle::ordered_map`, it hold a `std::unordered_map<int, vector<string>> dict_` and a `std::vector<std::string> names_`, we can find name from vector and get index to map the vector<string>, then replace all `VariableNameMap`.

I just think design like this is better, but the efficiency is not necessarily the highest

Reply:
Exactly, yes. It could be next work, to  the VariableNameMap with ordered_map.
Currently, I just want to enhance the ability of `getting-name-by-index` for several `contexts` in paddle. And, only with that the abstraction of common infershape fucntions, operators can be done.